src/main/scala/io/github/kiemlicz/shelm/HelmPlugin.scala

```diff
@@ -306,34 +379,125 @@ object HelmPublishPlugin extends AutoPlugin {
       val ivy = ivySbt.value
       new ivy.Module(moduleSettings.value)
     },
-    publishConfiguration := PublishConfiguration()
-      .withResolverName(Classpaths.getPublishTo(publishTo.value).name)
-      .withArtifacts(packagedArtifacts.value.toVector)
-      .withChecksums(checksums.value.toVector)
-      .withOverwrite(isSnapshot.value)
-      .withLogging(UpdateLogging.DownloadOnly),
+
+    publishHelmToIvyRepo := true,
+    outstandingPublishRequests := 1, //careful
+    publishToHosting := Seq.empty,
+
+    publish := Def.sequential(
+      Def.taskIf {
+        if (publishHelmToIvyRepo.value) {
+          streams.value.log.info("Starting Helm Charts publish to Ivy compatible repository")
+          /*
+          The 'original' SBT publish will use `publishConfiguration` settings, thus nothing else to be done here
+          */
+          (Helm / publish).value
+        } else
+          streams.value.log.info("No legacy Ivy-compatible repositories configured for publishing")
+      }.tag(Tags.Network, Tags.Publish), // affected: https://github.com/sbt/sbt/issues/6862 yet the publish is delegated to SBT's native publish which contains its own tags
+      Def.taskIf {
+        if (publishToHosting.value.exists(_.isInstanceOf[OciChartRegistry])) {
+          streams.value.log.info("Starting OCI login")
+          setupRegistries.value
+        } else
+          streams.value.log.info("No OCI registries configured for publishing")
+      },
+      Def.task {
+        val log = streams.value.log
+        log.info("Starting Helm Charts OCI or CM push")
+        /*
+        First fail causes stop of publish to given repo only (other registries are still tried)
+         */
+        val errors = publishToHosting.value.map {
+          case r: ChartMuseumRepository =>
+            chartMuseumClient.value.chartMuseumPublishBlocking(r, publishChartMuseumConfiguration.value, log)
+          case OciChartRegistry(uri, _) =>
+            sequence(
+              publishOCIConfiguration.value.artifacts.map {
+                case (_, file) => pushChart(file, uri, log)
+              }.toList
+            )
+          case unsupported =>
+            log.warn(s"Unsupported registry for publishing: $unsupported, omitting")
```

Comment:
This can log false-positives, iiuc.
Imagine `publishToHosting := [ivyRepo, cmRepo, ociReg]`

First task in the `Def.sequential` of `publish` will handle the `ivyRepo`.
Second will log in to the OCI registry.
Third will log that `ivyRepo` is unsupported and handle the rest.

The problem is that `ivyRepo` was correctly handled by the first task, and the warning log is a false-positive